src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,170 @@
+
+    }
+
+
+    /**
+     * Changes power to the arm
+     *
+     * @param power A double for the power going to the arm
+     */
+    fun setArmVelocity(power: Double) {
+        armState = ArmState.VELOCITY_CONTROL
+        masterSRX.set(ControlMode.Velocity, Math.abs(power))
+        armPower = power
+    }
+
+    private fun setArmPosition(position: Double) {
+        armAngle = position
+        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(position).toDouble())
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmVelocity(0.0)
+        brake.set(DoubleSolenoid.Value.kReverse)
+    }
+
+    /**
+     * Handles power to arm during each loop cycle
+     *
+     * @constructor Creates loop that controls power to arm during each loop cycle
+     */
+    val loop: Loop = object : Loop {
+
+        override fun onStart() {
+            movementState = MovementState.STATIONARY
+            setArmVelocity(0.0)
+            brake.set(DoubleSolenoid.Value.kForward)
+        }
+
+        /**
+         * Handles events during each loop cycle, depends on arm control mode
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                when (armState) {
+                    Arm.ArmState.LOW -> {
+                        setArmPosition(ArmState.LOW.targetPos)
+                    }
+                    Arm.ArmState.EXCHANGE -> {
+                        setArmPosition(ArmState.EXCHANGE.targetPos)
+                    }
+                    Arm.ArmState.HIGH -> {
+                        setArmPosition(ArmState.HIGH.targetPos)
+                    }
+                    //masterSRX.sensorCollection.quadratureVelocity
+                    Arm.ArmState.VELOCITY_CONTROL -> { }
+                }
+                if ((armState == Arm.ArmState.LOW && masterSRX.sensorCollection.quadratureVelocity > 0) || (armState == Arm.ArmState.HIGH && masterSRX.sensorCollection.quadratureVelocity < 0)) {
+                    movementState = Arm.MovementState.MOVING_TO_EXCHANGE
+                } else if (armState == Arm.ArmState.EXCHANGE && masterSRX.sensorCollection.quadratureVelocity > 0) {
+                    movementState = Arm.MovementState.MOVING_TO_TOP
+                } else if (armState == Arm.ArmState.EXCHANGE && masterSRX.sensorCollection.quadratureVelocity < 0) {
+                    movementState = Arm.MovementState.MOVING_TO_BOTTOM
+                } else if (masterSRX.sensorCollection.quadratureVelocity > 0) {
+                    movementState = Arm.MovementState.UP
+                } else if (masterSRX.sensorCollection.quadratureVelocity == 0) {
+                    stationaryTime ++
```

Comment:
`stationaryTime` is never actually set to zero, so the second time this happens it will instantly activate stationary. Also, stylistically the `++` shouldn't have a space before it.

Either way, it's probably better not to use just a counter but instead actually keep track of time elapsed; @plato2000 mentioned that there was some timing method that was used last year in the intake code if you want a reference point.